modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java

```diff
@@ -44,53 +43,72 @@
+        String gatewayUrl;
+        if (gatewayContextWrk.getContextTenant().getDomain().equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+            gatewayUrl = gatewayUrlsWrk.getWebAppURLNhttp();
+        } else {
+            gatewayUrl = gatewayUrlsWrk.getWebAppURLNhttp() + "t/" +
+                    gatewayContextWrk.getContextTenant().getDomain() + "/";
+        }
+        apiEndPointUrl = gatewayUrl + "jaxrs_basic/services/customers/customerservice";
+        APIRequest apiRequest = new APIRequest(API_NAME, API_CONTEXT, new URL(apiEndPointUrl));
+        apiRequest.setVersion(API_VERSION_1_0_0);
+        apiRequest.setSandbox(apiEndPointUrl);
+        apiRequest.setProvider(user.getUserName());
+        HttpResponse serviceResponse = restAPIPublisher.addAPI(apiRequest);
+        apiId = serviceResponse.getData();
     }
 
     @Test(groups = {"wso2.am"}, description = "Check custom life cycle state.")
     public void testCustomLifeCycle() throws Exception {
         //Create and publish api
-//        createAndPublishAPI(apiIdentifier, apiCreationRequestBean, apiPublisherClient, false);
-//        APILifeCycleStateRequest apiLifeCycleStatusChangeRequest = new APILifeCycleStateRequest(API_NAME,
-//                user.getUserName(), APILifeCycleState.PROMOTE);
-//        apiLifeCycleStatusChangeRequest.setVersion(apiIdentifier.getVersion());
-//
-//        //Change api status to custom state
-//        HttpResponse publishAPIResponse = apiPublisherClient.changeAPILifeCycleStatus(apiLifeCycleStatusChangeRequest);
-//        Boolean statusChangeSuccess = verifyAPIStatusChange(publishAPIResponse, APILifeCycleState.PUBLISHED,
-//                APILifeCycleState.PROMOTED);
-//        Assert.assertTrue(statusChangeSuccess, "API status Change is not successful");
+        restAPIPublisher.changeAPILifeCycleStatus(apiId, Constants.PUBLISHED);
+        Assert.assertEquals(APILifeCycleState.PUBLISHED.getState(), restAPIPublisher.getLifecycleStatus(apiId).
+                getData(), "lifecycle not changed to published");
+        restAPIPublisher.changeAPILifeCycleStatus(apiId, PROMOTE);
+        Assert.assertEquals(PROMOTE + "d", restAPIPublisher.getLifecycleStatus(apiId).getData(),
+                "lifecycle not changed to custom");
+        restAPIPublisher.changeAPILifeCycleStatus(apiId, RE_PUBLISH);
+        Assert.assertEquals(APILifeCycleState.PUBLISHED.getState(), restAPIPublisher.getLifecycleStatus(apiId).
+                        getData(),
+                "lifecycle not changed to published");
     }
 
     @AfterClass
     public void cleanupArtifacts() throws Exception {
         //Remove test api and revert to original lifecycle config
-        deleteAPI(apiIdentifier, apiPublisherClient);
-        lifeCycleAdminClient.editLifeCycle(apiLifeCycleName, originalLifeCycleContent);
-        loginClient.logOut();
-        apiPublisherClient.logout();
+        restAPIPublisher.deleteAPI(apiId);
```

Comment:
Lets have the cleanUp() at the end as a practice

Reply:
fixed